src/anemoi/inference/inputs/ekd.py

```diff
@@ -128,298 +106,61 @@ def __init__(
         self._namer = namer if namer is not None else self.checkpoint.default_namer()
         assert callable(self._namer), type(self._namer)
 
-    def _filter_and_sort(self, data: Any, *, variables: List[str], dates: List[Any], title: str) -> Any:
-        """Filter and sort the data.
+    def create_state(self, *, date=None, variables=None, initial: bool = True) -> State:
+        # NOTE: all logic involing the checkpoint should take place here
 
-        Parameters
-        ----------
-        data : Any
-            The data to filter and sort.
-        variables : List[str]
-            The list of variables to select.
-        dates : List[Any]
-            The list of dates to select.
-        title : str
-            The title for logging.
+        # define request (using checkpoint)
+        variables = self.checkpoint_variables if variables is None else variables
+        date = np.datetime64(date).astype(datetime.datetime)
```

Comment:
The logic of what happens when `date` is None has disappeared, I'm unable to run without specifying a date now.

When running from a grib file or dataset, it should pick the latest available date in the fieldlist. That logic used to be in the ekd.py  `_create_input_state`. When running from mars or cds, it should take yesterday's date.

You should also try to keep support for specifying the date like `-1` , which used to be handled by the `to_datetime()` utility.